weldx/measurement.py

```diff
@@ -37,71 +28,705 @@ class Signal:
+
+        """
+        if len(equipment.sources) > 1:
+            if source_name is None:
+                raise ValueError(
+                    "The equipment has multiple sources. Specify the "
+                    "desired one by providing a 'source_name'."
+                )
+            source = equipment.get_source(source_name)
+        elif len(equipment.sources) == 1:
+            source = equipment.sources[0]
+        else:
+            raise ValueError("The equipment does not have a source.")
+
+        mc = cls(name, source)
+        mc._source_equipment = equipment
+        return mc
+
+    @classmethod
+    def from_parameters(
+        cls,
+        name: str,
+        source_name: str,
+        source_error: Error,
+        output_signal_type: str,
+        output_signal_unit: str,
+        signal_data: "TimeSeries" = None,
+    ) -> "MeasurementChain":
+        """Create a new measurement chain without providing a `SignalSource` instance.
+
+        Parameters
+        ----------
+        name :
+            Name of the measurement chain
+        source_name :
+            Name of the source
+        source_error :
+            Error of the source
+        output_signal_type :
+            Type of the output signal ('analog' or 'digital')
+        output_signal_unit :
+            Unit of the output signal
+        signal_data :
+            Measured data of the sources' signal
+
+        Returns
+        -------
+        MeasurementChain :
+            New measurement chain
+
+        """
+        source = SignalSource(
+            source_name,
+            Signal(output_signal_type, output_signal_unit, signal_data),
+            source_error,
+        )
+        return cls(name, source)
+
+    def _add_signal(
+        self,
+        node_id: str,
+        signal: Signal = None,
+    ):
+        """Add a new signal node to the internal graph.
+
+        Parameters
+        ----------
+        node_id :
+            The name that will be used for the internal graphs' node. This is identical
+            to the name of the signals source.
+        signal :
+            The signal that should be added
+
+        """
+        self._raise_if_node_exist(node_id)
+
+        self._graph.add_node(node_id, signal=signal)
+        self._prev_added_signal = node_id
+
+    def _check_and_get_node_name(self, node_name: str) -> str:
+        """Check if a node is part of the internal graph and return its name.
+
+        If no name is provided, the name of the last added node is returned.
+
+        """
+        if node_name is None:
+            return self._prev_added_signal
+        self._raise_if_node_does_not_exist(node_name)
+        return node_name
+
+    @classmethod
+    def _determine_output_signal(
+        cls,
+        transformation: SignalTransformation,
+        input_signal: Signal,
+        data: "TimeSeries",
+    ) -> Signal:
+        """Get the signal that is produced by the provided transformation.
```

Comment:
```suggestion
        """Create a signal that is produced by the provided transformation.
```
The Signal doesn't exist in the graph/mc yet if I understand correctly?